tests/countries/test_micronesia.py

```diff
@@ -0,0 +1,538 @@
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_liberation_day(self):
+        name = "Liberation Day"
+        dt = (
+            "2004-09-10",
+            "2005-09-12",
+            "2010-09-10",
+            "2011-09-12",
+            "2016-09-12",
+            "2021-09-10",
+            "2022-09-12",
+        )
+        self.assertNoHolidayName(name)
+        for subdiv, holidays in self.subdiv_holidays.items():
+            if subdiv == "PNI":
+                self.assertHolidayName(
+                    name, holidays, (f"{year}-09-11" for year in range(1987, 2050))
+                )
+                self.assertHolidayName(f"{name} (observed)", holidays, dt)
+            else:
+                self.assertNoHolidayName(name, holidays)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_pohnpei_constitution_day(self):
+        name = "Pohnpei Constitution Day"
+        dt = (
+            "2003-11-07",
+            "2008-11-07",
+            "2009-11-09",
+            "2014-11-07",
+            "2015-11-09",
+            "2020-11-09",
+            "2025-11-07",
+        )
+        self.assertNoHolidayName(name)
+        for subdiv, holidays in self.subdiv_holidays.items():
+            if subdiv == "PNI":
+                self.assertHolidayName(
+                    name, holidays, (f"{year}-11-08" for year in range(1987, 2050))
+                )
+                self.assertHolidayName(f"{name} (observed)", holidays, dt)
+            else:
+                self.assertNoHolidayName(name, holidays)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_yap_day(self):
+        name = "Yap Day"
+        dt = (
+            "2003-02-28",
+            "2008-02-29",
+            "2009-03-02",
+            "2014-02-28",
+            "2015-03-02",
+            "2020-03-02",
+            "2025-02-28",
+        )
+        self.assertNoHolidayName(name)
+        for subdiv, holidays in self.subdiv_holidays.items():
+            if subdiv == "YAP":
+                self.assertHolidayName(
+                    name, holidays, (f"{year}-03-01" for year in range(1987, 2050))
+                )
+                self.assertHolidayName(f"{name} (observed)", holidays, dt)
+            else:
+                self.assertNoHolidayName(name, holidays)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_yap_state_constitution_day(self):
+        name = "Yap State Constitution Day"
+        dt = (
+            "2000-12-25",
+            "2005-12-23",
+            "2006-12-25",
+            "2011-12-23",
+            "2016-12-23",
+            "2017-12-25",
+            "2022-12-23",
+            "2023-12-25",
+        )
+        self.assertNoHolidayName(name)
+        for subdiv, holidays in self.subdiv_holidays.items():
+            if subdiv == "YAP":
+                self.assertHolidayName(
+                    name, holidays, (f"{year}-12-24" for year in range(1987, 2050))
+                )
```

Comment:
This is only holidays where we don't need for `assertNoNonObservedHoliday`, as Dec 25 coincides with Christmas.